### ISSUE TITLE
check_ssl_cert bug?

### ISSUE THREAD

**Issue by matteocorti** [matteocorti/check_ssl_cert#4]
I've been using check_ssl_cert for years with Nagios to monitor multiple hosts and it's working great! Recently however, I renewed an SSL cert for www.perturb.org, but check_ssl_cert is reporting that the SSL cert is expired. Somehow it still has the expiration date from the previous cert. The command Nagios runs is:

check_ssl_cert -H perturb.org --protocol https --port 443 --cn www.perturb.org --warning 14 --critical 7 --altnames

Which does indeed generate a critical alarm. All the browsers I've checked are happy, as are various SSL checking services:

https://www.sslshopper.com/ssl-checker.html#hostname=www.perturb.org
https://www.digicert.com/help/

Am I missing something, or is check_ssl_cert just confused?

**Comment by matteocorti** [comment]
I cannot reproduce it:

```
./check_ssl_cert -H perturb.org --protocol https --port 443 --cn www.perturb.org --warning 14 --critical 7 --altnames
SSL_CERT OK - X.509 certificate for 'www.perturb.org' from 'StartCom Class 1 Primary Intermediate Server CA' valid until Nov 18 06:11:24 2016 GMT (expires in 350 days)|days=350;14;7;;
```

Which version are you using?

**Comment by scottchiefbaker** [comment]
I think it may have been a config issue on my side. I updated the cert on www.perturb.org, but not perturb.org (which was just a redirect). I think it was a false alarm.

Is this script SNI aware?

**Comment by matteocorti** [comment]
Dear Scott

if you specify a CN with `--cn` and if your openssl supports the `-servername` the script will supply it. Should work but I did not really test it extensively.

Matteo
